NuRadioReco/detector/RNO_G/rnog_detector_mod.py

```diff
@@ -0,0 +1,251 @@
+
+        channel_id: int
+            The channel id
+
+        response: response.Response
+            A response object to be added to the `total_response`
+
+        """
+        orig_response = self.get_signal_chain_response(station_id, channel_id)
+        signal_chain_dict = self.get_channel_signal_chain(station_id, channel_id)
+
+        # modify the total response
+        signal_chain_dict["total_response"] = orig_response * response
+
+        # write the modified signal chain back to the buffered station
+        self._Detector__buffered_stations[station_id]["channels"][channel_id]['signal_chain'] = signal_chain_dict
+
+    def add_component(self, station_id, channel_id, component):
+        """ Add an additional component to the `response_chain` and the corresponding response to the `total_response`
+
+        Parameters
+        ----------
+        station_id: int
+            The station id
+
+        channel_id: int
+            The channel id
+
+        componennt: dict
+            A dictionary with the properties of the component to be added
+
+        """
+
+        # generate a response object from the component dict
+        component_response = Response(
+            component['frequencies'],
+            np.array([component['mag'], component['phase']]),
+            component['y-axis_units'],
+            time_delay=component['time_delay'],
+            name=component['name'],
+            station_id=station_id,
+            channel_id=channel_id
+    )
+
+        orig_response = self.get_signal_chain_response(station_id, channel_id)
+        signal_chain_dict = self.get_channel_signal_chain(station_id, channel_id)
+
+        # modify the signal chain
+        signal_chain_dict["total_response"] = orig_response * component_response
+        signal_chain_dict["response_chain"][component['name']] = component
+
+        # write the modified signal chain back to the buffered station
+        self._Detector__buffered_stations[station_id]["channels"][channel_id]['signal_chain'] = signal_chain_dict
+
+    def add_manual_time_delay(self, station_id, channel_id, time_delay, weight=1, name="MOD_manual_time_delay"):
+        """ Add an additional time delay to the signal chain and total response
+
+        Parameters
+        ----------
+        station_id: int
+            The station id
+
+        channel_id: int
+            The channel id
+
+        time_delay: float
+            The manual time delay to be added
+
+        weight: float (default: 1)
+            The weight of the time delay in the total response (either 1 or -1)
+
+        name: str (default: "MOD_manual_time_delay")
+            The name of the component to be added
+        """
+
+        component = {
+            'weight': weight,
+            'y-axis_units': ['mag', 'rad'],
+            'mag': [1, 1],
+            'phase': [0, 0],
+            'frequencies': [1e-3, 1e1],
+            'name': name,
+            'time_delay': time_delay
+        }
```

Comment:
@shallmann I think this is a bug. The database will return responses where the phase has a phase shift according to `time_delay`. This shift is removed in the constructor of the `Reponse` class. However, if the S21 parameter does not have this shift, it will be introduced and you double count the delay if you apply the response and change the trace start time.

Reply:
Fix it